application/models/services/SurveyUpdater/GeneralSettings.php

```diff
@@ -0,0 +1,570 @@
+<?php
+
+namespace LimeSurvey\Models\Services\SurveyUpdater;
+
+use Survey;
+use Permission;
+use LSYii_Application;
+use PluginEvent;
+use Date_Time_Converter;
+use LimeSurvey\SessionData;
+use LimeSurvey\PluginManager\PluginManager;
+use LimeSurvey\Models\Services\Exception\{
+    ExceptionPersistError,
+    ExceptionNotFound,
+    ExceptionPermissionDenied
+};
+
+/**
+ * Service GeneralSettings
+ *
+ * Service class for survey language setting updating.
+ *
+ * Dependencies are injected to enable mocking.
+ * @SuppressWarnings(PHPMD.ExcessiveClassComplexity)
+ */
+class GeneralSettings
+{
+    private Permission $modelPermission;
+    private Survey $modelSurvey;
+    private LSYii_Application $yiiApp;
+    private SessionData $sessionData;
+    private PluginManager $yiiPluginManager;
```

Comment:
I would not include `yii` here, since the plugin manager event system is actually _not_ based on the Yii events.

Reply:
Yeah this is a mistake. I didnt use the yii prefix for plugin manager elsewhere.